pkg/service/redisstore.go

```diff
@@ -242,21 +242,29 @@ func (s *RedisStore) LockRoom(_ context.Context, roomName livekit.RoomName, dura
 	return "", ErrRoomLockFailed
 }
 
-func (s *RedisStore) UnlockRoom(_ context.Context, roomName livekit.RoomName, uid string) error {
+func (s *RedisStore) UnlockRoom(ctx context.Context, roomName livekit.RoomName, uid string) error {
 	key := RoomLockPrefix + string(roomName)
+	unlockScript := `if redis.call("get", KEYS[1]) == ARGV[1] then
+						return redis.call("del", KEYS[1])
+					 else return 0 
+					 end`
+
+	res, err := redis.NewScript(unlockScript).Run(ctx, s.rc, []string{key}, uid).Result()
+	if err != nil {
+		if err == redis.Nil {
```

Comment:
redis.Nil isn't a valid response for the lua script, so it'd be better to remove this check.

Reply:
fixed